File: ts/src/cli/run-cli/emit-out-flag-notice.ts
import { extname } from "node:path";

import type { NormalizedCliOptions } from "./normalized-cli-options.js";

/**
 * Tell the user when `-o foo.json` would be silently treated as a
 * directory name. The dot in the basename is the heuristic for
 * "looks like a filename"; an empty extname means no notice.
 */
export function emitOutFlagNotice(opts: NormalizedCliOptions): void {
  if (opts.out === null || opts.out.mode !== "dir") {
    return;
  }
  if (extname(opts.out.path) === "") {
    return;
  }
  process.stderr.write(
    `uns: notice: -o '${opts.out.path}' is treated as a directory name; use --out-file to write to that path as a file.\n`,
  );
}

File: crates/unsnarl/src/cli/run_cli/emit_out_flag_notice.rs
use std::io::Write;
use std::path::Path;

/// Tell the user when `-o foo.json` would be silently treated as a
/// directory name. The dot in the basename is the heuristic for
/// "looks like a filename"; an empty extname means no notice.
pub fn emit_out_flag_notice(out_dir: Option<&Path>, stderr: &mut dyn Write) {
    let Some(path) = out_dir else {
        return;
    };
    if path.extension().is_none() {
        return;
    }
    let _ = writeln!(
        stderr,
        "uns: notice: -o '{}' is treated as a directory name; use --out-file to write to that path as a file.",
        path.display()
    );
}

#[cfg(test)]
#[path = "emit_out_flag_notice_test.rs"]
mod emit_out_flag_notice_test;
